homeassistant/components/ituran/strings.json

```diff
@@ -0,0 +1,45 @@
+{
+  "config": {
+    "step": {
+      "user": {
+        "data": {
+          "id_or_passport": "ID or passport number",
+          "phone_number": "Mobile phone number",
+          "mobile_id": "Mobile ID (leave empty if unknown)"
+        },
+        "data_description": {
+          "id_or_passport": "The goverment ID or passport number provided when registering with Ituran.",
+          "phone_number": "The mobile phone number provided when registering with Ituran. A one-time password will be sent to this mobile number.",
+          "mobile_id": "A unique ID used to identify the Ituran client. This can be shared between multiple clients to allow them all access to the Ituran services simultaniously. If one is not provided, it will be generated automatically."
```

Comment:
I'm not sure if I asked, but why do we ask the user this? What's the use case where people provide one?

Reply:
When using Ituran, if you "log in" from one device, it'll invalidate the previous one since it only saves one mobile ID per account. This means that if I use my mobile phone and then add the integration to HA, the mobile phone will be logged out. My way around this, to make them both work simultaneously, was to sniff the network traffic from my phone (which is how I learned how to use their web service in the first place) and save the mobile ID my phone generated. I don't think many people will do that but I didn't want to force myself, or anyone else, to go and edit HA's internal files in case they did want to do something like or have multiple HA instances that work together, etc.

There might be  more deterministic way to know what the phone's mobile ID is but I wasn't able to decrypt their iOS app to disassemble it and try to how they generate it.

Reply:
I discussed this in the core team, and having the user require sniffing network traffic from a mobile app in order to setup the integration is not an experience we want to provide to our user, even not as an optional step.

Reply:
I get that, but what about having multiple HA instances sharing the same ID? Is that not a valid use-case?
I can always rename it from "Mobile ID" to something else, that's just the parameter name of their web service.

If I remove it completely then I won't be able to add the integration while sharing the ID since I need it to complete authentication in the config flow before it's added. Adding it with a different (random) ID, authenticating, manually editing the HA entity config file, manually authentication in a different setup (at least until reauth is added) and restating HA sound very cumbersome (IMHO) as opposed to an optional field that most users will see only.

Again, your call but it doesn't sound that bad to me. Let me know if I should remove it altogether.

Reply:
I think we in general expect users to have only one Home Assistant instance, and I think this would be a good known limitation to mention in the docs. I can check this with the core team, but that might take a little time. If you remove the field, we can merge this in core and discuss this in a followup PR and then I can check with the core team afterwards. WDYT?

Reply:
OK, I'll remove it from the forms but it will need to stay in the configuration. Can you just let me know what's expected for the reauth flow? Can I immediately send a request for an OTP once the user clicks on the reauthenticate button or do I need some confirmation message before that?

Reply:
Why do we have to keep it in the configuration? 

How long is the OTP code valid?

Reply:
The mobile ID is basically the password for accessing the web service so, once it's registered, I need to save it persistently. The OPT is really one-time and is not saved.
The registered mobile ID seems to be valid for 1 month

Reply:
Oh right, that is true, good point, yep that is true.

Maybe for reauth it is nice to have a step in between that it will send the user an SMS or email or whatever they use to let them know and then ask them

Reply:
Done, removed the mobile ID field from the form